config_loader_test.go

```diff
@@ -45,8 +51,27 @@ func Test_loadFromEnvVars_errIfNotPointer(t *testing.T) {
 	assert.Error(t, err, "expected error if settings not a pointer")
 }
 
+func Test_matchEnvVarToField(t *testing.T) {
+	settings := &TestSettings{}
+	os.Setenv("PORT", "5000")
```

Comment:
You might like [`t.Setenv`](https://pkg.go.dev/testing#T.Setenv), which will clean this up.